format-delimited/src/main/java/io/cdap/plugin/format/delimited/input/PathTrackingDelimitedInputFormat.java

```diff
@@ -101,22 +111,34 @@ public NullWritable getCurrentKey() {
                 throw new IOException(message + " Did you mean to use the 'text' format?");
               }
             }
-            throw new IOException(message + " Check that the schema contains the right number of fields.");
+            if (!enableQuotesValue && delimitedString.contains(QUOTE)) {
+              message += " Check if quoted values should be allowed.";
+            }
+            throw new IOException(
+              message + " Check that the schema contains the right number of fields.");
           }
 
           Schema.Field nextField = fields.next();
           if (part.isEmpty()) {
             builder.set(nextField.getName(), null);
           } else {
             String fieldName = nextField.getName();
-            //Ensure if date time field, value is in correct format
+            // Ensure if date time field, value is in correct format
             SchemaValidator.validateDateTimeField(nextField.getSchema(), fieldName, part);
             builder.convertAndSet(fieldName, part);
           }
         }
         return builder;
       }
 
+      private Iterator<String> getSplitsIterator(boolean enableQuotesValue, String delimitedString, String delimiter) {
+        if (!enableQuotesValue) {
```

Comment:
nit: a little more readable to do `if (enableQuotesValue)` instead of the opposite.